homeassistant/components/devolo_home_network/switch.py

```diff
@@ -0,0 +1,141 @@
+        entity_registry_enabled_default=True,
+        icon="mdi:wifi",
+        name="Enable guest Wifi",
+        is_on_func=lambda data: data.enabled is True,
+        turn_on_func=lambda device: device.device.async_set_wifi_guest_access(True),  # type: ignore[union-attr]
+        turn_off_func=lambda device: device.device.async_set_wifi_guest_access(False),  # type: ignore[union-attr]
+    ),
+    SWITCH_LEDS: DevoloSwitchEntityDescription[bool](
+        key=SWITCH_LEDS,
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=True,
+        icon="mdi:led-off",
+        name="Enable LEDs",
+        is_on_func=bool,
+        turn_on_func=lambda device: device.device.async_set_led_setting(True),  # type: ignore[union-attr]
+        turn_off_func=lambda device: device.device.async_set_led_setting(False),  # type: ignore[union-attr]
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Get all devices and sensors and setup them via config entry."""
+    device: Device = hass.data[DOMAIN][entry.entry_id]["device"]
+    coordinators: dict[str, DataUpdateCoordinator[Any]] = hass.data[DOMAIN][
+        entry.entry_id
+    ]["coordinators"]
+
+    entities: list[DevoloSwitchEntity[Any]] = []
+    if device.device and "led" in device.device.features:
+        entities.append(
+            DevoloSwitchEntity(
+                entry,
+                coordinators[SWITCH_LEDS],
+                SWITCH_TYPES[SWITCH_LEDS],
+                device,
+            )
+        )
+    if device.device and "wifi1" in device.device.features:
+        entities.append(
+            DevoloSwitchEntity(
+                entry,
+                coordinators[SWITCH_GUEST_WIFI],
+                SWITCH_TYPES[SWITCH_GUEST_WIFI],
+                device,
+            )
+        )
+    async_add_entities(entities)
+
+
+class DevoloSwitchEntity(DevoloEntity[_DataT], SwitchEntity):
+    """Representation of a devolo switch."""
+
+    entity_description: DevoloSwitchEntityDescription[_DataT]
+
+    def __init__(
+        self,
+        entry: ConfigEntry,
+        coordinator: DataUpdateCoordinator[_DataT],
+        description: DevoloSwitchEntityDescription[_DataT],
+        device: Device,
+    ) -> None:
+        """Initialize entity."""
+        self.entity_description = description
+        super().__init__(entry, coordinator, device)
+
+    @property
+    def is_on(self) -> bool:
+        """State of the switch."""
+        return self.entity_description.is_on_func(self.coordinator.data)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the entity on."""
+        try:
+            await self.entity_description.turn_on_func(self.device)
+        except DevicePasswordProtected:
+            require_reauth(self.hass, self.entry)
+        except DeviceUnavailable:
+            pass  # The coordinator will handle this
```

Comment:
Are we missing coverage here?

Reply:
If the device is unabailable, the next coordinator update will fail with the same exception and that case is covered by a test. Shall I test it here again?

Reply:
All cases should preferably be tested. It looked like you tested this case, so I wasn't sure if codecov was correct.